client.go

```diff
@@ -2909,3 +2807,121 @@ func (c *pipelineConnClient) PendingRequests() int {
 }
 
 var errPipelineConnStopped = errors.New("pipeline connection has been stopped")
+
+var DefaultTransport RoundTripper = &Transport{}
+
+type Transport struct{}
+
+func (t *Transport) RoundTrip(hc *HostClient, req *Request, resp *Response) (retry bool, err error) {
```

Comment:
```suggestion
var DefaultTransport RoundTripper = &transport{}

type transport struct{}

func (t *transport) RoundTrip(hc *HostClient, req *Request, resp *Response) (retry bool, err error) {
```
I don't think we need to expose this. We already have a huge API surface so I would like to keep this private.